tests/test_gen.py

```diff
@@ -0,0 +1,85 @@
+from happytransformer import HappyGeneration
+from happytransformer import default_greedy_settings, default_p_nucleus_sampling_settings, default_beam_settings, default_top_k_sampling_settings, default_generic_sampling_settings
+
+def test_default_simple():
+    happy_gen = HappyGeneration()
+    output = happy_gen.generate_text("I went to the basement and then", min_length=5, max_length=40)
+    assert type(output.text) == str
+
+def test_default_min_max_length():
+    happy_gen = HappyGeneration()
+    output = happy_gen.generate_text("I went to the basement and then", min_length=5, max_length=5)
+    tokens = happy_gen.tokenizer.encode(output.text, return_tensors="pt")
+    length = len(tokens[0])
+    assert length == 5
+
+
+def test_all_methods():
+    happy_gen = HappyGeneration()
+    output_greedy = happy_gen.\
+        generate_text("I went to the basement and then",
+                      settings=default_greedy_settings, min_length=5, max_length=20)
+
+    output_beam_search = happy_gen.\
+        generate_text("I went to the basement and then",
+                      settings=default_p_nucleus_sampling_settings, min_length=5, max_length=20)
+
+    output_generic_sampling = happy_gen.\
+        generate_text("I went to the basement and then",
+                      settings=default_beam_settings, min_length=5, max_length=20)
+
+    output_top_k_sampling = happy_gen.\
+        generate_text("I went to the basement and then",
+                      settings=default_top_k_sampling_settings, min_length=5, max_length=20)
+
+    output_top_p_nucleus_sampling = happy_gen.\
+        generate_text("I went to the basement and then",
+                      settings=default_generic_sampling_settings, min_length=5, max_length=20)
```

Comment:
backslashes are generally considered bad style. It could instead be written as so.

```
output_top_p_nucleus_sampling = happy_gen.generate_text(
    "I went to the basement and then",
    settings=default_generic_sampling_settings, min_length=5, max_length=20
)
```

Reply:
Fixed in a1ada70